### ISSUE TITLE
Population Based training with tf.Estimator

### ISSUE THREAD

**Issue by agniszczotka** [tensorflow/tensorflow#21737]
OS Platform and Distribution N/A
TensorFlow installed from N/A
TensorFlow version 1.10
Bazel version N/A
CUDA/cuDNN version N/A
GPU model and memory N/A
Exact command to reproduce N/A
Mobile device N/A

Feature request:

Can you implement population-based training with tf.Estimator? 
https://deepmind.com/blog/population-based-training-neural-networks/

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by agniszczotka** [comment]
I have updated my feature request

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ispirmustafa: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
Thanks @agniszczotka -- that is an interesting area of research. While we don't currently have resources to devote to a pre-baked solution here, we are happy to consider contributions that enable population based training for Estimators-- let us know if you are interested.

**Comment by Squadrick** [comment]
@karmel I'd love to take a crack at this. Any pointers before I get started?

**Comment by karmel** [comment]
Great-- anything that can also work for Keras would be preferred-- so, if you represent additional functionality as layers/models that then become part of a canned Estimator, we have much wider applicability and flexibility. Thanks, and looking forward to updates--

**Comment by Squadrick** [comment]
Alright, I'll get started right away. I'm creating a PR to track progress.

**Comment by agniszczotka** [comment]
Please find a toy [example](https://github.com/agniszczotka/PBT-TF.Estimator-with-Ray.Tune/blob/master/trainable_estimator_with_pbt.py) showing how to use tf.Estimator with [Ray.Tune implementation of PBT](https://ray.readthedocs.io/en/latest/tune-schedulers.html#population-based-training-pbt).

**Comment by sushreebarsa** [comment]
@agniszczotka 
Is this still an issue ?
We see that you are using TF v1.x which is not actively supported and we recommend you to upgrade latest TF versions(2.4 or later).Please refer to the [migration](https://www.tensorflow.org/guide/migrate) doc .
Estimators are not recommended for new code. Estimators  can behave unexpectedly, especially when combined with TF 2 code. Could you please refer to this [link](https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier) for more details  and let us know if it helps?
Thanks!
